pkg/kubelet/prober/prober.go

```diff
@@ -142,9 +147,27 @@ func (pb *prober) runProbe(ctx context.Context, probeType probeType, p *v1.Probe
 		return pb.exec.Probe(pb.newExecInContainer(ctx, container, containerID, command, timeout))
 
 	case p.HTTPGet != nil:
-		req, err := httpprobe.NewRequestForHTTPGetAction(p.HTTPGet, &container, status.PodIP, "probe")
-		if err != nil {
-			return probe.Unknown, "", err
+		var err error
+		var req *http.Request
+		key := probeKey{podUID: pod.UID, containerName: container.Name, probeType: probeType}
+		reqAny, found := httpProbeReqCache.Load(key)
+		// cache miss, create request object
+		if !found {
+			req, err = httpprobe.NewRequestForHTTPGetAction(p.HTTPGet, &container, status.PodIP, "probe")
+			if err != nil {
+				return probe.Unknown, "", err
+			}
+			httpProbeReqCache.Store(key, req)
+		} else {
+			req = reqAny.(*http.Request)
+		}
+		// cache hit, but pod IP changed, recreate the request object
+		if p.HTTPGet.Host == "" && req.URL.Hostname() != status.PodIP {
+			req, err = httpprobe.NewRequestForHTTPGetAction(p.HTTPGet, &container, status.PodIP, "probe")
+			if err != nil {
+				return probe.Unknown, "", err
+			}
+			httpProbeReqCache.Store(key, req)
```

Comment:
Do I understand correctly that in case of cache miss this is repeated twice?
```
req, err = httpprobe.NewRequestForHTTPGetAction(p.HTTPGet, &container, status.PodIP, "probe")
if err != nil {
	return probe.Unknown, "", err
}
httpProbeReqCache.Store(key, req)
```

Would it make sense to combine `!found` and this scope?

Reply:
```
Would it make sense to combine !found and this scope?
```
Done